docs/configuration.md

```diff
@@ -0,0 +1,128 @@
+  for all requests.
+- `String googleCredentialsPath` - path to file containing GoogleCredentials
+  json. May be set with the JDBC Connection
+  Property `cloudSqlGoogleCredentialsPath`.
+
+Users should only set one field. If more than one field is set, the connector 
+will use the credential from the first field with a value in this order of 
+precedence: googleCredentialSupplier, googleCredentials, googleCredentialsPath.
+
+## Configuration Reference
+
+### Connection Configuration
+
+These properties configure the connection:
+
+| JDBC Property Name          | R2DBC Property Name | Description                                                                                                                                                                                                                                | Default Value    | Example                           |
+|-----------------------------|---------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------|-----------------------------------|
+| cloudSqlInstance (required) | HOST                | Identifies the instance to connect                                                                                                                                                                                                         |                  | `projectname:region:instancename` |
+| cloudSqlNamedConnector      | NAMED_CONNECTOR     | References a connector configuration created using `ConnectorRegistry.register()`                                                                                                                                                          |                  | `my-configuration`                |
+| unixSocketPath              | UNIX_SOCKET         | The path to the unix socket to connect to the database, when using the java connector with the [Cloud SQL Auth Proxy](https://github.com/GoogleCloudPlatform/cloud-sql-proxy/)                                                             |                  | `/var/db/my-db-instance`          |
+| enableIamAuth               | ENABLE_IAM_AUTH     | Enable IAM Authentication to authenticate to the database. Valid values: `true` - authenticate with the IAM principal,  `false` - authenticate with a database user and password.                                                          | false            | `true`                            |
+| ipTypes                     | IP_TYPES            | A comma-separated list of IP types, ordered by preference. Value values: `PUBLIC` - connect to the instance's public IP, `PRIVATE` - connect to the instances private IP, `PSC` - connect to the instance through Private Service Connect. | `PUBLIC,PRIVATE` | `PSC,PRIVATE,PUBLIC`              |
+
+### Connector Configuration
```

Comment:
Let's start with Connector. Since the flow goes create the connector -> configure the connection -> connect.

Reply:
Fixed.